ext/curl/interface.c

```diff
@@ -757,6 +757,54 @@ static size_t curl_xferinfo(void *clientp, curl_off_t dltotal, curl_off_t dlnow,
 /* }}} */
 #endif
 
+#if LIBCURL_VERSION_NUM >= 0x075400
+static size_t curl_ssh_hostkeyfunction(void *clientp, int keytype, const char *key, size_t keylen)
+{
+	php_curl *ch = (php_curl *)clientp;
+	php_curl_callback *t = ch->handlers.sshhostkey;
+	size_t rval = 0;
+
+#if PHP_CURL_DEBUG
+	fprintf(stderr, "curl_ssh_hostkeyfunction() called\n");
+	fprintf(stderr, "clientp = %x, keytype = %d, key = %s, keylen = %zu\n", clientp, keytype, key, keylen);
+#endif
+
+	zval argv[4];
+	zval retval;
+	zend_result error;
+	zend_fcall_info fci;
+
+	GC_ADDREF(&ch->std);
+	ZVAL_OBJ(&argv[0], &ch->std);
+	ZVAL_LONG(&argv[1], keytype);
+	ZVAL_STRING(&argv[2], key);
+	ZVAL_LONG(&argv[3], keylen);
+
+	fci.size = sizeof(fci);
+	ZVAL_COPY_VALUE(&fci.function_name, &t->func_name);
+	fci.object = NULL;
+	fci.retval = &retval;
+	fci.param_count = 4;
+	fci.params = argv;
+	fci.named_params = NULL;
+
+	ch->in_callback = 1;
+	error = zend_call_function(&fci, &t->fci_cache);
+	ch->in_callback = 0;
+	if (error == FAILURE) {
+		php_error_docref(NULL, E_WARNING, "Cannot call the CURLOPT_SSH_HOSTKEYFUNCTION");
+	} else if (!Z_ISUNDEF(retval)) {
```

Comment:
You're pointing to the documentation of `CURLOPT_SSH_KEYFUNCTION` instead of [`CURLOPT_SSH_HOSTKEYFUNCTION`](https://curl.se/libcurl/c/CURLOPT_SSH_HOSTKEYFUNCTION.html).